`PREPARE GROUPING AS SHOW CHANGEFEED JOB error` does not round-trip

From https://github.com/cockroachdb/cockroach/issues/65475#issuecomment-849033451:

sql/tests.TestRandomSyntaxGeneration failed with artifacts on master @ 4e3437fb275f1eb7645ca04d942c82a596c3826e:

```
E210526 18:16:35.480930 2217438 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 71  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:16:38.104175 2246618 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 72  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
    rsg_test.go:736: 3m50s of 5m0s: 893916 executions, 10859 successful
E210526 18:16:39.736545 2263557 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 73  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:16:42.362867 2292033 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 74  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:16:44.136650 2311891 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 75  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
    rsg_test.go:736: 3m55s of 5m0s: 915561 executions, 11146 successful
E210526 18:16:45.449242 2326202 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 76  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:16:47.928047 2353311 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 77  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
    rsg_test.go:736: 4m0s of 5m0s: 937679 executions, 11447 successful
E210526 18:16:49.746443 2372327 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 78  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:16:51.524150 2392291 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 79  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/244: cannot split range at key /Table/3/1/244
E210526 18:16:53.470556 2413006 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 80  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/244: cannot split range at key /Table/3/1/244
E210526 18:16:53.874395 2417991 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 81  making txn commit explicit failed for "unnamed" meta={id=164f3909 pri=0.07767077 epo=0 ts=1622053013.872612766,0 min=1622053013.872612766,0 seq=2} lock=true stat=STAGING rts=1622053013.872612766,0 wto=false gul=1622053014.372612766,0 ifw=1: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:736: 4m5s of 5m0s: 958901 executions, 11693 successful
E210526 18:16:58.884088 2424418 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 82  making txn commit explicit failed for "unnamed" meta={id=7bd9ce24 pri=0.02160966 epo=0 ts=1622053018.880677797,0 min=1622053018.880677797,0 seq=2} lock=true stat=STAGING rts=1622053018.880677797,0 wto=false gul=1622053019.380677797,0 ifw=1: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:736: 4m10s of 5m0s: 961481 executions, 11719 successful
    rsg_test.go:736: 4m15s of 5m0s: 966414 executions, 11767 successful
    rsg_test.go:736: 4m20s of 5m0s: 988557 executions, 12051 successful
    rsg_test.go:736: 4m25s of 5m0s: 1010314 executions, 12327 successful
E210526 18:17:15.532402 2563346 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 83  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:17:17.905118 2589973 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 84  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:17:18.759125 2599038 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 85  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:17:19.476421 2606459 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 86  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
    rsg_test.go:736: 4m30s of 5m0s: 1032067 executions, 12574 successful
E210526 18:17:22.049557 2634276 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 87  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
    rsg_test.go:736: 4m35s of 5m0s: 1053613 executions, 12844 successful
E210526 18:17:24.568804 2661271 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 88  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:17:25.764083 2674578 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 89  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:17:27.884671 2700644 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 90  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:17:29.510695 2719887 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 91  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
    rsg_test.go:736: 4m40s of 5m0s: 1077055 executions, 13180 successful
E210526 18:17:31.652655 2744632 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 92  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:17:33.498221 2766717 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 93  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
    rsg_test.go:736: 4m45s of 5m0s: 1100086 executions, 13486 successful
E210526 18:17:35.765792 2792682 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 94  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:17:37.564879 2814103 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 95  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
    rsg_test.go:736: 4m50s of 5m0s: 1124089 executions, 13811 successful
E210526 18:17:39.543059 2838098 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 96  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/246: cannot split range at key /Table/3/1/246
E210526 18:17:41.827567 2864099 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 97  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/292: cannot split range at key /Table/3/1/292
E210526 18:17:44.100464 2890049 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 98  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/292: cannot split range at key /Table/3/1/292
    rsg_test.go:736: 4m55s of 5m0s: 1147446 executions, 14129 successful
E210526 18:17:45.782589 2909644 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 99  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/292: cannot split range at key /Table/3/1/292
E210526 18:17:45.978648 2912170 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 100  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/292: cannot split range at key /Table/3/1/292
E210526 18:17:47.511219 2930133 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 101  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/292: cannot split range at key /Table/3/1/292
    rsg_test.go:772: 1170546 executions, 14433 successful
    rsg_test.go:780: cannot parse output of Format: sql="PREPARE GROUPING AS SHOW CHANGEFEED JOB error", formattedSQL="PREPARE \"grouping\" AS SHOW CHANGEFEED JOBVALUES (error)": at or near "jobvalues": syntax error
    rsg_test.go:237: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRandomSyntaxGeneration357254424
--- FAIL: TestRandomSyntaxGeneration (300.17s)
```

--- a/docs/generated/sql/bnf/show_jobs.bnf
+++ b/docs/generated/sql/bnf/show_jobs.bnf
@@ -5,6 +5,7 @@ show_jobs_stmt ::=
 	| 'SHOW' 'JOBS' select_stmt
 	| 'SHOW' 'JOBS' 'WHEN' 'COMPLETE' select_stmt
 	| 'SHOW' 'JOBS' for_schedules_clause
+	| 'SHOW' 'CHANGEFEED' 'JOBS' select_stmt
 	| 'SHOW' 'JOB' job_id
 	| 'SHOW' 'CHANGEFEED' 'JOB' job_id
 	| 'SHOW' 'JOB' 'WHEN' 'COMPLETE' job_id

--- a/docs/generated/sql/bnf/stmt_block.bnf
+++ b/docs/generated/sql/bnf/stmt_block.bnf
@@ -657,6 +657,7 @@ show_jobs_stmt ::=
 	| 'SHOW' 'JOBS' select_stmt
 	| 'SHOW' 'JOBS' 'WHEN' 'COMPLETE' select_stmt
 	| 'SHOW' 'JOBS' for_schedules_clause
+	| 'SHOW' 'CHANGEFEED' 'JOBS' select_stmt
 	| 'SHOW' 'JOB' a_expr
 	| 'SHOW' 'CHANGEFEED' 'JOB' a_expr
 	| 'SHOW' 'JOB' 'WHEN' 'COMPLETE' a_expr

--- a/pkg/sql/parser/sql.y
+++ b/pkg/sql/parser/sql.y
@@ -5091,6 +5091,10 @@ show_jobs_stmt:
   {
     $$.val = &tree.ShowJobs{Schedules: $3.slct()}
   }
+| SHOW CHANGEFEED JOBS select_stmt
+  {
+    $$.val = &tree.ShowChangefeedJobs{Jobs: $4.slct()}
+  }
 | SHOW JOBS select_stmt error // SHOW HELP: SHOW JOBS
 | SHOW JOB a_expr
   {

--- a/pkg/sql/parser/testdata/show
+++ b/pkg/sql/parser/testdata/show
@@ -466,6 +466,22 @@ EXPLAIN SHOW USERS -- literals removed
 EXPLAIN SHOW USERS -- identifiers removed
 
 parse
+SHOW JOB 1234
+----
+SHOW JOBS VALUES (1234) -- normalized!
+SHOW JOBS VALUES ((1234)) -- fully parenthesized
+SHOW JOBS VALUES (_) -- literals removed
+SHOW JOBS VALUES (1234) -- identifiers removed
+
+parse
+EXPLAIN SHOW JOB 1234
+----
+EXPLAIN SHOW JOBS VALUES (1234) -- normalized!
+EXPLAIN SHOW JOBS VALUES ((1234)) -- fully parenthesized
+EXPLAIN SHOW JOBS VALUES (_) -- literals removed
+EXPLAIN SHOW JOBS VALUES (1234) -- identifiers removed
+
+parse
 SHOW JOBS
 ----
 SHOW JOBS
@@ -496,6 +512,22 @@ EXPLAIN SHOW AUTOMATIC JOBS
 EXPLAIN SHOW AUTOMATIC JOBS -- fully parenthesized
 EXPLAIN SHOW AUTOMATIC JOBS -- literals removed
 EXPLAIN SHOW AUTOMATIC JOBS -- identifiers removed
+
+parse
+SHOW CHANGEFEED JOB 1234
+----
+SHOW CHANGEFEED JOBS VALUES (1234) -- normalized!
+SHOW CHANGEFEED JOBS VALUES ((1234)) -- fully parenthesized
+SHOW CHANGEFEED JOBS VALUES (_) -- literals removed
+SHOW CHANGEFEED JOBS VALUES (1234) -- identifiers removed
+
+parse
+EXPLAIN SHOW CHANGEFEED JOB 1234
+----
+EXPLAIN SHOW CHANGEFEED JOBS VALUES (1234) -- normalized!
+EXPLAIN SHOW CHANGEFEED JOBS VALUES ((1234)) -- fully parenthesized
+EXPLAIN SHOW CHANGEFEED JOBS VALUES (_) -- literals removed
+EXPLAIN SHOW CHANGEFEED JOBS VALUES (1234) -- identifiers removed
 
 parse
 SHOW CHANGEFEED JOBS

--- a/pkg/sql/sem/tree/show.go
+++ b/pkg/sql/sem/tree/show.go
@@ -294,12 +294,10 @@ type ShowChangefeedJobs struct {
 
 // Format implements the NodeFormatter interface.
 func (node *ShowChangefeedJobs) Format(ctx *FmtCtx) {
-	ctx.WriteString("SHOW CHANGEFEED")
+	ctx.WriteString("SHOW CHANGEFEED JOBS")
 	if node.Jobs != nil {
-		ctx.WriteString(" JOB")
+		ctx.WriteString(" ")
 		ctx.FormatNode(node.Jobs)
-	} else {
-		ctx.WriteString(" JOBS")
 	}
 }
 

